minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/RunMiNiFi.java

```diff
@@ -1562,6 +1564,12 @@ void setAutoRestartNiFi(final boolean restart) {
     }
 
     void setMiNiFiCommandControlPort(final int port, final String secretKey) throws IOException {
+
+        if (this.secretKey != null && this.ccPort != UNINITIALIZED_CC_PORT) {
```

Comment:
Just wondering if these will always be set together, or would `||` be more appropriate?